frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1039,20 +1046,20 @@ class SegmentsView extends React.Component<Props, State> {
     const segmentGroupToChangeColor = this.getSegmentsOfGroup(groupId);
     if (segmentGroupToChangeColor == null) return;
 
-    const actionArray = segmentGroupToChangeColor.map((segment) =>
+    const actions = segmentGroupToChangeColor.map((segment) =>
       updateSegmentAction(segment.id, { color: color }, visibleSegmentationLayer.name),
     );
 
-    Store.dispatch(batchUpdateGroupsAndSegmentsAction(actionArray));
+    Store.dispatch(batchUpdateGroupsAndSegmentsAction(actions));
   }
 
   handleRefreshMeshes = (groupId: number) => {
     const { visibleSegmentationLayer } = this.props;
     if (visibleSegmentationLayer == null) return;
-    const segmentGroup = this.getSegmentsOfGroup(groupId);
-    if (segmentGroup == null) return;
+    //TODO I didnt find a better way to call the visiblesegmentationlayer. one other way would be to pass it into the function. but that's not useful for functions below-
+    // so my solution is a little more code duplication. putting it higher up isn't really a solution.
```

Comment:
I'm not sure what you mean by "call the visiblesegmentationlayer" ? Accessing it via `const { visibleSegmentationLayer } = this.props;` seems fine to me if that's what you are asking about?

Reply:
right, accessing would be the right word here. 
In `handlePerSegment` I do the exact check as in these lines, and I was wondering whether I could circumvent this check right here.

Reply:
ah I see. since the repeated lines include a return-statement which influences the control flow, I don't see a simple way to DRY this. Therefore, I think it's fine to keep it as is :) So, you can remove the todo comment.